cmd/serverless/main.go

```diff
@@ -164,6 +165,9 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 	// some useful warnings first
 
 	var apikeySetIn = []string{}
+	if os.Getenv(kmsAPIKeyEnvVarLegacy) != "" && os.Getenv(kmsAPIKeyEnvVar) == "" {
```

Comment:
Change looks good, do you think we could get a test for this?

Reply:
I'm talking with Alex about it, think it will have to be in the self monitoring app, since the KMS setup is a bit difficult to replicate in a unit test.